app/src/main/java/com/google/samples/apps/sunflower/viewmodels/PlantDetailViewModel.kt

```diff
@@ -36,6 +40,29 @@ class PlantDetailViewModel(
     val isPlanted: LiveData<Boolean>
     val plant: LiveData<Plant>
 
+    /**
+     * This is the job for all coroutines started by this ViewModel.
+     *
+     * Cancelling this job will cancel all coroutines started by this ViewModel.
+     */
+    private val viewModelJob = Job()
+
+    /**
+     * This is the scope for all coroutines launched by [PlantDetailViewModel].
+     *
+     * Since we pass [viewModelJob], you can cancel all coroutines launched by [viewModelScope] by calling
+     * viewModelJob.cancel().  This is called in [onCleared].
+     */
+    private val viewModelScope = CoroutineScope(IO + viewModelJob)
```

Comment:
Prefer to show `Dispatchers.Main` here and have the repository use `withContext` to switch to the IO thread. This makes the repository always safe to call from any dispatcher, instead of exporting a dispatcher dependency as part of it's API.